native/src/hhRoutePlanner.cpp

```diff
@@ -1267,4 +1266,82 @@ bool HHRoutePlanner::retainAll(std::set<int64_t> & source, const std::set<int64_
     }
 }
 
+void HHRoutePlanner::filterPointsBasedOnConfiguration(SHARED_PTR<HHRoutingContext> & hctx) {
+    SHARED_PTR<GeneralRouter>  & vr = hctx->rctx->config->router;
+    UNORDERED_map<string, RoutingParameter> & parameters = vr->getParameters();
+    UNORDERED_map<string, string> tm;
+    for (const auto & es : vr->getParameterValues()) {
+        string paramId = es.first;
+        // These parameters don't affect the routing filters
+        // This assumption probably shouldn't be used in general and only for popular parameters)
+        if (parameters.find(paramId) != parameters.end() && paramId != GeneralRouterConstants::USE_SHORTEST_WAY
+            && paramId != GeneralRouterConstants::USE_HEIGHT_OBSTACLES
+            && paramId.rfind(GeneralRouterConstants::GROUP_RELIEF_SMOOTHNESS_FACTOR, 0) != 0) {
```

Comment:
Why rfind()? Just find() should be enough.

Reply:
find() will return first position - 0, therefore better to rfind

Reply:
I think it doesn't matter
 paramId.startsWith(GeneralRouter.GROUP_RELIEF_SMOOTHNESS_FACTOR) is the same find. However they both not correct translation of startsWith by the internla algorithm.

Chatgpt says: ``` 
bool startsWith(const std::string& paramId, const std::string& prefix) {
    if (paramId.length() < prefix.length()) {
        return false; // paramId is shorter than prefix, cannot start with prefix
    }
    return paramId.substr(0, prefix.length()) == prefix;
}
```

Which I think more correct startsWith